services/src/test/scala/cromwell/services/instrumentation/impl/stackdriver/StackdriverInstrumentationServiceActorSpec.scala

```diff
@@ -0,0 +1,111 @@
+package cromwell.services.instrumentation.impl.stackdriver
+
+import akka.testkit.{TestActorRef, TestProbe}
+import cats.data.NonEmptyList
+import com.typesafe.config.ConfigFactory
+import cromwell.core.TestKitSuite
+import cromwell.services.instrumentation.InstrumentationService.InstrumentationServiceMessage
+import cromwell.services.instrumentation._
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{FlatSpecLike, Matchers}
+
+import scala.concurrent.duration._
+
+class StackdriverInstrumentationServiceActorSpec extends TestKitSuite with FlatSpecLike with Matchers with Eventually {
+  behavior of "StackdriverInstrumentationServiceActor"
+
+  val MaxWaitTime = 2.minutes
+  implicit val pc: PatienceConfig = PatienceConfig(MaxWaitTime)
+
+  val globalConfig = ConfigFactory.parseString(
+    s"""
+       |google {
+       |  application-name = "cromwell"
+       |  auths = [
+       |    {
+       |      name = "application-default"
+       |      scheme = "application_default"
+       |    }
+       |  ]
+       |}
+      """.stripMargin
+  )
+
+  val registryProbe = TestProbe().ref
+  val resourceLabels = Map("project_id" -> "my-project")
+
+  val testBucket = CromwellBucket(List("test_prefix"), NonEmptyList.of("test", "metric", "bucket"))
+  val testGaugeBucket = CromwellBucket(List("test_prefix"), NonEmptyList.of("test", "gauge", "metric", "bucket"))
+
+
+  it should "correctly receive the metrics with resource labels" in {
+    val stackdriverConfig = ConfigFactory.parseString(
+      """
+        |auth = "application-default"
+        |google-project = "my-project"
+        |flush-rate = 1 minute
+      """.stripMargin
+    )
+
+    val rawMetricList = List(
+      CromwellIncrement(testBucket),
+      CromwellCount(testBucket, 80),
+      CromwellGauge(testGaugeBucket, 20),
+      CromwellTiming(testBucket.expand("timing"), 5.seconds),
+      CromwellIncrement(testBucket),
+      CromwellCount(testBucket, 80),
+      CromwellGauge(testGaugeBucket, 60),
+      CromwellTiming(testBucket.expand("timing"), 10.seconds)
+    )
+
+    val receivedCumulativeMetrics = ("custom.googleapis.com/cromwell/test_prefix/test/metric/bucket", 162.0)
```

Comment:
is this `expectedReceivedCumulativeMetrics`?

Reply:
Yes. I can rename it to `expectedCumulativeMetrics` if it makes more sense?